lib/components/search_bar.dart

```diff
@@ -41,8 +41,7 @@ class _SearchBarState extends State<SearchBar> {
   void didUpdateWidget(SearchBar oldWidget) {
     super.didUpdateWidget(oldWidget);
     // If query param has changed
-    if (oldWidget.initialQuery != widget.initialQuery &&
-        widget.initialQuery != null) {
+    if (widget.initialQuery != null) {
```

Comment:
I think you still need the `oldWidget.initialQuery != widget.initialQuery &&` in case the widget updates for another reason